### ISSUE TITLE
[TASK][EASY] Add a common getSessionConf method in spark engine module

### ISSUE THREAD

**Issue by wForget** [apache/kyuubi#5438]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/296) and found no similar tasks.

### Mentor

- [X] I have sufficient knowledge and experience of this task, and I volunteer to be the mentor of this task to guide contributors to complete the task.

### Skill requirements

- Knowledge of Kyuubi
- Familiar with the open-source ecosystem

### Background and Goals

As discussed in https://github.com/apache/kyuubi/pull/5410#discussion_r1360164253, some session-level configurations are ignored due to the complexity of get session-level configurations in kyuubi spark engine, so we can implement a common method to get session-level configurations.

### Implementation steps

Add a common method in `org.apache.kyuubi.engine.spark.KyuubiSparkUtil`, like:

```
  def getSessionConf[T](configEntry: ConfigEntry[T], spark: SparkSession): T = {
    spark.conf.getOption(configEntry.key).map(configEntry.valueConverter).getOrElse {
      SparkSQLEngine.kyuubiConf.get(configEntry)
    }
  }
```

Refactor the code that gets configurations using `getSessionConf` method.

**Comment by davidyuan1223** [comment]
i think i can try improvement this continue then fix the #5410's test case

**Comment by wForget** [comment]
> i think i can try improvement this, then fix the #5410's test case

thanks, assigned
